File: oj/web/src/store/problem.js
export default {
    state: {
        id: "",
        number: "",
        title: "",
        type: "",
        score: "",
        topicDescription: "",
        inputDescription: "",
        outputDescription: "",
        exampleOneInput: "",
        exampleOneOutput: "",
        exampleTwoInput: "",
        exampleTwoOutput: "",
        exampleThreeInput: "",
        exampleThreeOutput: "",
        link: "",
        competitionId: "",
        competitionName: "",
        competitionDetails: "",
        competitionProblemNumber: "",
    },
    getters: {},
    mutations: {
        updateProblem(state, problem) {
            state.id = problem.id;
            state.number = problem.number;
            state.title = problem.title;
            state.type = problem.type;
            state.topicDescription = problem.topicDescription;
            state.inputDescription = problem.inputDescription;
            state.outputDescription = problem.outputDescription;
            state.exampleOneInput = problem.exampleOneInput;
            state.exampleOneOutput = problem.exampleOneOutput;
            state.exampleTwoInput = problem.exampleTwoInput;
            state.exampleTwoOutput = problem.exampleTwoOutput;
            state.exampleThreeInput = problem.exampleThreeInput;
            state.exampleThreeOutput = problem.exampleThreeOutput;
            state.link = problem.link;
        },
        updateCompetitionProblem(state,problem){
            state.id = problem.id;
            state.number = problem.number;
            state.title = problem.title;
            state.type = problem.type;
            state.score = problem.score;
            state.topicDescription = problem.topicDescription;
            state.inputDescription = problem.inputDescription;
            state.outputDescription = problem.outputDescription;
            state.exampleOneInput = problem.exampleOneInput;
            state.exampleOneOutput = problem.exampleOneOutput;
            state.exampleTwoInput = problem.exampleTwoInput;
            state.exampleTwoOutput = problem.exampleTwoOutput;
            state.exampleThreeInput = problem.exampleThreeInput;
            state.exampleThreeOutput = problem.exampleThreeOutput;
        },
        removeProblem(state){
            state.id = "";
            state.number = "";
            state.title = "";
            state.type = "";
            state.topicDescription = "";
            state.inputDescription = "";
            state.outputDescription ="";
            state.exampleOneInput = "";
            state.exampleOneOutput = "";
            state.exampleTwoInput = "";
            state.exampleTwoOutput = "";
            state.exampleThreeInput = "";
            state.exampleThreeOutput = "";
            state.link = "";
        },
        setCompetitionId(state,competitionId){
            state.competitionId=competitionId;
        },
        setCompetitionName(state,competitionName){
            state.competitionName=competitionName;
        },
        setCompetitionDetails(state,competitionDetails){
            state.competitionDetails=competitionDetails;
        },
        setCompetitionProblemNumber(state,competitionProblemNumber){
            state.competitionProblemNumber = competitionProblemNumber;
        }
    },
    modules: {}
}

File: oj/backclound/backend/src/main/java/com/oj/backend/pojo/Problem.java
package com.oj.backend.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Problem {
    @TableId(type = IdType.AUTO)
    private Integer id;
    private Integer number;
    private String title;
    private String type;
    private String numberTitleType;
    private String topicDescription;
    private String inputDescription;
    private String outputDescription;
    private String exampleOneInput;
    private String exampleOneOutput;
    private String exampleTwoInput;
    private String exampleTwoOutput;
    private String exampleThreeInput;
    private String exampleThreeOutput;
    private String link;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date modifyTime;
}
